ingestors/kafka/src/main/scala/hydra/kafka/model/TopicMetadataV2Transport.scala

```diff
@@ -95,27 +105,25 @@ final case class TopicMetadataV2Request(
 }
 
 object TopicMetadataV2Request {
+
   type NumPartitionsPredicate = Greater[W.`9`.T] And Less[W.`51`.T]
   type NumPartitions = Int Refined NumPartitionsPredicate
+
   object NumPartitions extends RefinedTypeOps[NumPartitions, Int]
-  type SubjectRegex = MatchesRegex[W.`"""^(?:skills|flow|tech|fin|dvs|_[a-zA-Z0-9]+)\\.[a-zA-Z0-9\\-\\.]+"""`.T]
+
+  val regex = s"^(?=^.{0,255}$$)(?:$AllowedOrganizations)(\\.[a-zA-Z0-9]+(\\-[a-zA-Z0-9]+)*)+"
```

Comment:
Can you give an explanation of what this regex is doing?

Reply:
`(?=^.{0,255}$$)` - checks max subject length
`(?:$AllowedOrganizations)` - checks subject prefixes
`(\\.[a-zA-Z0-9]+(\\-[a-zA-Z0-9]+)*)+` - checks each group (like `.group1.group2.group-3` and so on).
             |                     ||
   `\\.[a-zA-Z0-9]+` -  checks that each group contains at least 1 letter or number after `.` sign
                                   ||
                        (\\-[a-zA-Z0-9]+)*  - checks that each group contains 0 or more parts with '-' sign and at least 1 letter or 
                                                        number after it.

Using this regex we can avoid additional checks we had:
`if(value.length > 255 ||
        value.contains(".-") ||
        value.contains("-.") ||
        value.contains("..") ||
        value.contains("--")) `